### PR TITLE
4 Sum problem

### PR DESCRIPTION
**Pull request by kotwani2883**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added 4 Sum Problem

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2480

### CODE REVIEW

**Review by iamrajiv — APPROVED** [review]

**Comment by kotwani2883** [comment]
I have resolved the conflicts kindly give it a check.

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -280,8 +280,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)]
```
Comment (siddharth25pandey) [inline C-Plus-Plus/README.md]: Remove one line (Duplication of same line)

File: C-Plus-Plus/other/4Sum.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+					return ;
+				}
+			}
+		}
+	}
+}
+
+
+int main() {
+	int n;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp]: Write good naming convention

File: C-Plus-Plus/other/4Sum.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	cin >> n;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp:44]: Display a message in print form as what are you taking input from the user

File: C-Plus-Plus/other/4Sum.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	int sum;
+	cin >> sum;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp:47]: Display a message in print form as what are you taking input from the user

File: C-Plus-Plus/other/4Sum.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	int a[n];
+	for (int i = 0; i < n; i++)
+		cin >> a[i];
+	findFourElements(a, n, sum);
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp]: Display a message in print form as what are you displaying output to the user

File: C-Plus-Plus/other/4Sum.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	}
+}
+
+
+int main() {
+	int n;
+	cin >> n;
+	int sum;
+	cin >> sum;
+	int a[n];
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp]: Write good naming convention

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kotwani2883 Do the changes as suggested

**Comment by kotwani2883** [comment]
I have made the changes.Kindly give it a check.

File: C-Plus-Plus/README.md [hunk 7]
```diff
@@ -371,7 +369,4 @@ _add list here_
 - [KnapSack Fractional using Greedy Approach](other/knapsackfraction.cpp)
 - [Longest Subarray with equal number of 0 and 1](other/Longest_subarray_with_equal_0_1.cpp)
 - [Set Clear and Check bits](other/check_set_clear_bit.cpp)
-- [Iterative approach to print all subsets](other/subsets_iterative.cpp)
-- [Count of Set Bits in an Integer using Lookup Technique](other/count_of_set_bit_using_lookup_tech.cpp)
-- [Concatenate given strings](other/n_string_concatenate.cpp)
-- [Trapping Rainwater](other/trappingRainwater.cpp)
```
Comment (siddharth25pandey) [inline C-Plus-Plus/README.md]: unnecessary deletion of the previous content,
Add it.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kotwani2883 Minor changes is required

**Comment by kotwani2883** [comment]
Kindly Check. I have made the changes.
Thanks!!

**Review by siddharth25pandey — APPROVED** [review]
@kotwani2883 LGTM, able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @kotwani2883 for her code

**Pull request merged**